runtime/ops/os/mod.rs

```diff
@@ -68,7 +68,22 @@ fn noop_op() -> Result<(), AnyError> {
 
 #[op]
 fn op_exec_path(state: &mut OpState) -> Result<String, AnyError> {
-  let current_exe = env::current_exe().unwrap();
+  // Rust has platform-dependent symlink resolution, so we need to be clever
+  // if we want to get the called path to the executable.
+  // See:
+  // https://github.com/rust-lang/rust/issues/43617
+  // https://github.com/rivy/rs.coreutils/commit/c457dfbbc469333e7f116971b610a8e3fa1cdf66
+  let current_exe = match std::env::args().next() {
+    Some(ref s) if !s.is_empty() => {
+      let argv0 = std::path::PathBuf::from(s);
+      if argv0.is_absolute() {
+        argv0
+      } else {
+        std::env::current_dir().unwrap().join(argv0)
```

Comment:
I think this will break if a user does `Deno.chdir` before calling this function. I wonder if we should follow Rust and leave this up to users to do? I'm not sure what's best.

Reply:
Hm. That's a valid concern. I'll take a look at the execution environment and see if we can detect that.

Obviously, continuing to do what Rust does is one type of expectable behavior. It's unfortunate that different kernels do this differently. So, I'm torn between it's surprising because it's different from Rust or it's surprising because of environmental inconsistencies.

Obviously, we could use `canonicalize()` and always resolve the symlinks, but that feels like the worst version.

Reply:
```rust
use std::env;

fn main() {
  println!("{}", env::current_dir().unwrap().to_str().unwrap());
  env::set_current_dir("..").unwrap();
  println!("{}", env::current_dir().unwrap().to_str().unwrap());
}
```

Obviously does what you'd expect: "/Users/jacob\n/Users". And given that Deno.chdir() calls ops_chdir() >  set_current_dir(), we can assume that your case will do that. So, a proper (full) solution would be to compute the value from argv[0] at invocation time, and cache it for the duration.

Reply:
Other options could include skipping it for relative executions paths, or passing those out instead of resolving to absolute paths.